faillint/faillint.go

```diff
@@ -11,28 +11,39 @@ import (
 	"golang.org/x/tools/go/analysis"
 )
 
-// Analyzer of the linter
-var Analyzer = &analysis.Analyzer{
-	Name:             "faillint",
-	Doc:              "report unwanted import path usages",
-	Run:              run,
-	RunDespiteErrors: true,
-}
+type faillint struct {
+	paths       string // -paths flag
+	ignoretests bool   // -ignoretests flag
 
-var paths string // -paths flag
+}
 
-func init() {
-	// seems like using init() is the only way to add our own flags
-	Analyzer.Flags.StringVar(&paths, "paths", paths, "import paths to fail")
+// NewAnalyzer create a faillint analyzer
+func NewAnalyzer() *analysis.Analyzer {
+	f := faillint{
+		paths:       "",
+		ignoretests: false,
+	}
+	a := &analysis.Analyzer{
+		Name:             "faillint",
+		Doc:              "report unwanted import path usages",
+		Run:              f.run,
+		RunDespiteErrors: true,
+	}
+	a.Flags.StringVar(&f.paths, "paths", "", "import paths to fail")
+	a.Flags.BoolVar(&f.ignoretests, "ignoretests", false, "ignore all _test.go files and packages.")
```

Comment:
```suggestion
	a.Flags.BoolVar(&f.ignoretests, "ignore-tests", false, "ignore all _test.go files and packages.")i 
```

I think this is better than `ignoretests`